qiime2/core/type/signature.py

```diff
@@ -268,7 +268,10 @@ def _assert_valid_outputs(self, outputs):
     def decode_parameters(self, **kwargs):
         params = {}
         for key, spec in self.parameters.items():
-            params[key] = spec.qiime_type.decode(kwargs[key])
+            if spec.has_default() and spec.default is kwargs[key] is None:
```

Comment:
@ebolyen, I am with @jairideout on that one: can you update to his proposed form?